addons/account/tests/test_ir_actions_report.py

```diff
@@ -96,3 +97,141 @@ def test_download_one_encrypted_pdf(self):
+        """
+        first_invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'date': '2024-01-01',
+            'invoice_date': '2024-01-01',
+            'partner_id': self.partner_a.id,
+            'invoice_line_ids': [(0, 0, {
+                'name': 'Something',
+                'quantity': 1,
+                'price_unit': 123,
+                'product_id': self.product_a.id
+            })]
+        })
+
+        second_invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'date': '2024-01-01',
+            'invoice_date': '2024-01-01',
+            'partner_id': self.partner_a.id,
+            'invoice_line_ids': [(0, 0, {
+                'name': 'Something',
+                'quantity': 1,
+                'price_unit': 123,
+                'product_id': self.product_a.id
+            })]
+        })
+
+        third_invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'date': '2024-01-01',
+            'invoice_date': '2024-01-01',
+            'partner_id': self.partner_a.id,
+            'invoice_line_ids': [(0, 0, {
+                'name': 'Something',
+                'quantity': 1,
+                'price_unit': 123,
+                'product_id': self.product_a.id
+            })]
+        })
+
+        # Post invoices to be able to associate attachments.
+        for invoice in [first_invoice, second_invoice, third_invoice]:
+            invoice.action_post()
+
+        invoices_report_ref = 'account.report_invoice_with_payments'
+
+        # Generate report for second invoice to create an attachment.
+        second_invoice_report_content, content_type = self.env['ir.actions.report'].with_context(
+            force_report_rendering=True)._render_qweb_pdf(invoices_report_ref, res_ids=second_invoice.id)
+        self.assertEqual(content_type, "pdf", "Report is not a PDF")
+        self.assertTrue(second_invoice_report_content, "PDF not generated")
+
+        # Make sure the attachment is created.
+        report = self.env['ir.actions.report']._get_report(invoices_report_ref)
+        self.assertTrue(report.attachment, f"Report '{invoices_report_ref}' doesn't save attachments")
+        self.assertTrue(report.retrieve_attachment(second_invoice), "Attachment not generated")
+
+        aggregate_report_content, content_type = self.env['ir.actions.report'].with_context(
+            force_report_rendering=True)._render_qweb_pdf(invoices_report_ref,
+                                                          res_ids=[first_invoice.id, second_invoice.id,
+                                                                   third_invoice.id])
+        self.assertEqual(content_type, "pdf", "Report is not a PDF")
+        self.assertTrue(aggregate_report_content, "PDF not generated")
+
+    def test_report_with_some_resources_reloaded_from_attachment_with_multiple_page_invoice(self):
```

Comment:
Do we really need a specific test for this ? Why not doing both in the same ?

Reply:
Yes, each test is for a different logic flow in the code. The code generates PDFs in batch so it needs to split them afterward for each invoice. This is done differently based on whether each invoice has one page ([code](https://github.com/odoo/odoo/blob/4bb5f2462a65b97bc4bfd507cfc1a4fde5f6b836/odoo/addons/base/models/ir_actions_report.py#L753)), or at least one has multiple pages ([code](https://github.com/odoo/odoo/blob/4bb5f2462a65b97bc4bfd507cfc1a4fde5f6b836/odoo/addons/base/models/ir_actions_report.py#L767)). The fix was done for both.

Reply:
I would still appreciate the assertions to be factorized, or to use sub-tests

[Keep it DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)